posts/2021-11-26-jakarta-ee-9.1.adoc

```diff
@@ -0,0 +1,429 @@
+    private String field1;
+    public String field2;
+}
+----
++
+results in
++
+[source, yaml]
+----
+  schemas:
+    Example:
+      type: object
+      properties:
+        field1:
+          type: string
+        field2:
+          type: string
+----
++
+The field `field1` should not have appeared in the generated OpenAPI document as it is private. This issue has been fixed by setting the `mp.openapi.extensions.smallrye.private-properties.enable` property to disable non-public properties by default.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/19057[Port bind skipped at server startup]
++
+Previously, in an extremely rare scenario, configured ports could silently fail to bind - preventing Liberty from using them. This issue was caused by a subtle race condition in the code responsible for delaying the port bind until the server is ready to handle traffic.
++
+In the failing scenario, the port started message would not be emitted - for example the following message would be missing:
++
+`CWWKO0219I: TCP Channel defaultHttpEndpoint has been started and is now listening for requests on host *  (IPv4) port 9080.`
++
+and the following FFDC will be seen:
++   
+[source, no_copy]
+----
+Exception = java.lang.RuntimeException
+Source = com.ibm.ws.tcpchannel.internal.NBAcceptChannelSelector
+probeid = 254
+Stack Dump = java.lang.RuntimeException: java.nio.channels.NotYetBoundException
+	at com.ibm.ws.tcpchannel.internal.NBAcceptChannelSelector.performRequest(NBAcceptChannelSelector.java:358)
+	at com.ibm.ws.tcpchannel.internal.ChannelSelector.run(ChannelSelector.java:257)
+	at java.base/java.lang.Thread.run(Thread.java:866)
+Caused by: java.nio.channels.NotYetBoundException
+	at java.base/sun.nio.ch.ServerSocketChannelImpl.begin(ServerSocketChannelImpl.java:246)
+	at java.base/sun.nio.ch.ServerSocketChannelImpl.accept(ServerSocketChannelImpl.java:283)
+	at com.ibm.ws.tcpchannel.internal.NBAcceptChannelSelector.performRequest(NBAcceptChannelSelector.java:278)
+	... 2 more
+----
++
+This issue has now been fixed so that all configured ports should start, or if there is a problem some meaningful error message should be logged.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18992[Application fails to restart in server.xml update scenario]
++
+We discovered an issue where an application would fail to restart, due to a race condition during server reconfiguration when multiple apps are installed. The problem occurs when one app starts before another app is finished uninstalling. In theory this shouldn't be a problem - however for this scenario these apps are sharing a VirtualHost configuration object, and in this case one app updates the parent VirtualHost as part of its uninstall process in such a way that the other gets into an invalid state. The server log will show an warning such as `CWWKZ0020I: Application <app_name> not updated`. This issue was fixed by fixing the race condition that caused the failure.
```

Comment:
The server log will show an warning
--->
The server log will show a warning

Reply:
```suggestion
We discovered an issue where an application would fail to restart, due to a race condition during server reconfiguration when multiple apps are installed. The problem occurs when one app starts before another app is finished uninstalling. In theory this shouldn't be a problem - however for this scenario these apps are sharing a VirtualHost configuration object, and in this case one app updates the parent VirtualHost as part of its uninstall process in such a way that the other gets into an invalid state. The server log will show a warning such as `CWWKZ0020I: Application <app_name> not updated`. This issue was fixed by fixing the race condition that caused the failure.
```